### ISSUE TITLE
Feature request: Ability to insert R code snippets

### ISSUE THREAD

**Issue by drleehw** [jasp-stats/jasp-issues#137]
The pace of development with JASP is very impressive, and it's now to the point where it's usable for many analyses, but there will always remain a great many more specialised algorithms one might wish to run. In most packages, e.g. SPSS, this can be done in syntax, so that the analysis can all be run in one environment, and the output is all together in one place. 

Could JASP please have the ability to put add R code snippets to the output, and have them execute? I see it working something like this:

1. Press the R Code Snippet button on the menu.
2. A blank output space appears on the right, and an R scripting window appears on the left. 
3. User types or pastes in some R code on the left.
4. JASP executes the code much as RStudio does, and displays all output, text and graphical, in the given space on the right. 
5. The code snippet remains editable if one clicks back on the relevant section in the output. 

This could also be a nice solution for manipulating variables for power users so long as R code snippets in R were had the ability to alter data files. 

I know it's a big request, but it would be a big step in JASP getting to the point where I could use it to teach postgrad stats. I'm sure many users would be very very grateful.

**Comment by juliusverrel** [comment]
Any news on this?

**Comment by jspickering** [comment]
I know you've got a statement about this on your website (https://jasp-stats.org/faq/jasp-uses-r-analyses-possible-export-r-code-used-analysis/) but I was wondering if there's any more info on when there might be an update where we can view/copy the underlying R code from our output?

**Comment by TimKDJ** [comment]
I can't give you an exact time frame, but it's definitely a few months out still. It'll be my #1 priority once 0.12 releases (in about two weeks).

**Comment by jspickering** [comment]
Great, thank you! JASP has been a lifesaver so far, keep up the good work 😄

**Comment by TimKDJ** [comment]
Glad to hear it :)

**Comment by tomtomme** [comment]
We kind of have this now, do we?

**Comment by boutinb** [comment]
Yes, more features on R Syntax will come, but we can definitely closes this issue.
